### ISSUE TITLE
existing implementations - merger or rewrite?

### ISSUE THREAD

**Issue by bill-auger** [LiveCodingTVOfficial/PHP-API-Wrapper#1]
it looks like LCTV has taken an interest in community projects - in particular a PHP API wrapper of which there are three people i know that have each already created an implementation

there has been some (off-topic) discussion toward consolidating the codebases in another thread --> https://github.com/Wapaca/livecoding-auth/issues/7 so i propose to continue that conversation here 

@Wapaca --> https://github.com/Wapaca/livecoding-auth/
@andtrev --> https://github.com/andtrev/lctv-badges/blob/master/php/LCTVAPI.php
@OzanKurt --> https://github.com/OzanKurt/LiveCoding-API

**Comment by OzanKurt** [comment]
In my opinion it would be nice to keep 1 repository with composer and one repository that is just one file. It would be easier to include in non-composer projects. When it comes to my repository, I am thinking about removing file storage since I don't really think that people would use it while there is session.

**Comment by andtrev** [comment]
How were you planning on using sessions? I don't see them as very useful for an API library, as it limits access, unless there's something I'm missing. They're fine for the authorization process. I guess you're thinking the app using the library will store the tokens and cache info on its own?

**Comment by bill-auger** [comment]
yes - i mentioned in [wapaca's README](https://github.com/Wapaca/livecoding-auth) that the base library only supports volatile sessions and primitive flat file storage and that anyone with more sophisticated use-cases should subclass the LivecodingAuthTokens class for their purpose - i imagined that we would add at the most one additional reference subclass implementation for SQLITE

**Comment by bill-auger** [comment]
also i noticed that someone has forked [Wapaca's repo](https://github.com/Wapaca/livecoding-auth) into this organization which reminds me that DrM mentioned they would like to announce the HTML badges soon - he has some more ideas for designs and has offered hosting once we have the project in good shape so they can probably get a pretty url like badges.livecoding.tv - i would like to move the auth business out of the badges project and link it as a submodule but we still need to decide which implementation to use - once this is sorted out i will move the badges project into this organization like: https://github.com/LiveCodingTVOfficial/lctv-badges and link it to the API auth library

@Wapaca -
@andtrev -
it would be nice if we could arrange a pair-programming session soon to merge your PHP API auth implementations into this repo or into the [livecoding-auth fork](https://github.com/LiveCodingTVOfficial/livecoding-auth) and delete one of them - even without my help i think that you two could merge the best parts of the two implementations with less effort than i could on my own

**Comment by andtrev** [comment]
My implementation already supports MySQL. We really need to talk about what the project is supposed to achieve, as my implementation goes beyond the simple online/offline static button, I have 5 badges atm, Streaming Status, Live Viewers, Number of Followers, Last Streamed, and Next Stream, these all display dynamic content retrieved from the API. The design is the standard buttons you'll find on github or http://shields.io/ - https://github.com/andtrev/lctv-badges

I'm free week nights after 8PST and anytime weekends to schedule a get together.

**Comment by bill-auger** [comment]
the original feature request for the HTML badges was a bit vague but suggested there could be may variants showing any of the available stats - i made the online status just as a seed to get started because that is something i had planned to make for myself anyway - but those we should discuss on the badges tracker

my reason for posting here was because i would like to separate the API interactions into its own project - that is presumable why this repo was created -  and to decide if your implementation is fully featured and should be in this repo or if wapaca's project and yours each have some capability that the other does not then to merge the best parts together into a single library that is fully featured - and then in either case to link the API library to the badges project as a submodule

**Comment by andtrev** [comment]
Yes, you mentioned the badges so I thought I'd kill two birds with one stone, but that should probably be talked about elsewhere.

The main difference between mine and wapaca's implementation is the storage code, I abstract getting, setting and deleting data generically so it encompasses auth and API calls, where as his looks to be specifically based around the authorization process, but I think you (Bill) wrote that part so you would know better. As I said before, I have a class for MySQL and one for flat files, SQLite or anything else would be easy enough to add. I guess the question is he much wrapping of the API do we want? Right now we basically have none, just a very simple REST client, which I'm perfectly happy with. Doing this in person will probably be easier, which I'm open to, but the others seem busy, which is understandable, all of us are donating our time.

**Comment by bill-auger** [comment]
if you are online now this may be a good time then - i just started setting up the badges repo on a staging server  - LCTV is broken tonight but at least the chat is working - im idling in my LCTV chat if ou want to dro pin https://www.livecoding.tv/bill-auger/chat

**Comment by andtrev** [comment]
Sorry I missed this, I followed you in lctv now so it should email me whenever you're on and I'll try and hop on so we can chat, unless you want to schedule something for the future.
